app/assets/v2/js/grants/detail.js

```diff
@@ -73,68 +74,87 @@ $(document).ready(function() {
-          '</div>' +
-          '<div class="col-12 mt-4 mb-2 text-right font-caption">' +
-            '<a rel="modal:close" href="javascript:void" aria-label="Close dialog" class="button button--primary-o mr-3">' + gettext('No, I don\'t want to cancel') + '</a>' +
-            '<button class="modal-cancel-grants button button--warning">' + gettext('Cancel this Grant') + '</button>' +
-          '</div>' +
-        '</div>' +
-      '</div>'
-    );
-    var modal = $(content).appendTo('#js-cancel_grant').modal({
+    $('#cancelModal').modal({
       modalClass: 'modal cancel_grants'
     });
 
     $('.modal-cancel-grants').on('click', function(e) {
       let contract_address = $('#contract_address').val();
+      let grant_cancel_tx_id;
       let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, contract_address);
 
       web3.eth.getAccounts(function(err, accounts) {
         deployedSubscription.methods.endContract()
-          .send({from: accounts[0], gasPrice: 4000000000})
-          .on('transactionHash', function() {
-            document.issueURL = document.getElementById('form--input__reference-url').value;
+          .send({
+            from: accounts[0],
+            gas: 3000000,
+            gasPrice: web3.utils.toHex($('#gasPrice').val() * Math.pow(10, 9))
+          }).on('transactionHash', function(transactionHash) {
+            grant_cancel_tx_id = $('#grant_cancel_tx_id').val();
+            const linkURL = etherscan_tx_url(transactionHash);
+
+            document.issueURL = linkURL;
+            $('#transaction_url').attr('href', linkURL);
             $('.modal .close').trigger('click');
             enableWaitState('#grants-details');
           })
           .on('confirmation', function(confirmationNumber, receipt) {
             $.ajax({
               type: 'post',
               url: '',
-              data: { 'contract_address': contract_address},
+              data: {
+                'contract_address': contract_address,
+                'grant_cancel_tx_id': grant_cancel_tx_id
+              },
               success: function(json) {
                 window.location.reload(false);
               },
               error: function() {
-                alert('Canceling you grant failed to save. Please try again.');
+                alert('Canceling your grant failed to save. Please try again.');
               }
             });
           });
       });
     });
   });
 
+  $('#contract_owner_button').on('click', function(e) {
+    let contract_owner_address = $('#contract_owner_address').val();
+    let contract_address = $('#contract_address').val();
+    let grant_cancel_tx_id;
+    let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, contract_address);
+
+    web3.eth.getAccounts(function(err, accounts) {
+      deployedSubscription.methods.changeOwnership(
+        contract_owner_address
+      ).send({
+        from: accounts[0],
+        gasPrice: 8000000000
```

Comment:
gas price should not be hardcoded -- why not `web3.utils.toHex($('#gasPrice').val() * Math.pow(10, 9)`  ?

Reply:
Need to add gasSettings to owner change as well.